src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -86,32 +55,29 @@ static uct_cuda_ipc_dev_cache_t *uct_cuda_ipc_create_dev_cache(int dev_num)
     return cache;
 }
 
-static uct_cuda_ipc_dev_cache_t *uct_cuda_ipc_get_dev_cache(const CUuuid *uuid)
+static uct_cuda_ipc_dev_cache_t *
+uct_cuda_ipc_get_dev_cache(uct_cuda_ipc_component_t *component,
+                           const CUuuid *uuid)
 {
-    uct_cuda_ipc_dev_cache_t *cache   = NULL;
-    khash_t(cuda_ipc_uuid_hash) *hash = &uct_cuda_ipc_uuid_cache.hash;
+    khash_t(cuda_ipc_uuid_hash) *hash = &component->uuid_hash;
+    uct_cuda_ipc_dev_cache_t *cache;
     khiter_t iter;
     int ret;
 
     iter = kh_put(cuda_ipc_uuid_hash, hash, *uuid, &ret);
-    switch (ret) {
-    case UCS_KH_PUT_BUCKET_EMPTY:
-    case UCS_KH_PUT_BUCKET_CLEAR:
+    if (ret == UCS_KH_PUT_KEY_PRESENT) {
+        cache = kh_val(hash, iter);
```

Comment:
return kh_val(hash, iter);

we can add return in each if/else and eliminate the last return and make the code a bit simpler to follow

Reply:
ok